src/features/profile/pages/NotificationSettings/NotificationsSettings.tsx

```diff
@@ -50,9 +50,9 @@ export const NotificationsSettings = () => {
   const hasUserChanged = !!user?.subscriptions && hasUserChangedParameters(user, state)
 
   const updatePushPermissionFromSettings = (permission: PermissionStatus) => {
-    if (permission === 'granted' && !state.allowPush) {
-      dispatch({ type: 'push' })
-    } else if (permission !== 'granted' && state.allowPush) dispatch({ type: 'push' })
+    if (permission === 'granted' && user?.subscriptions.marketingPush) {
+      dispatch({ type: 'push', state: true })
+    } else if (permission !== 'granted') dispatch({ type: 'push', state: false })
```

Comment:
Pourquoi cette condition ? Est-ce que ça suffit pas de faire ça : 
```suggestion
    } else dispatch({ type: 'push', state: false })
```